File: demos/iOS/Example-iOS-ObjC/Example-iOS-ObjC/SimplePaymentTransactionController.m
//
//  SimplePaymentTransactionController.m
//  Example-iOS-ObjC
//
//  Created by Hildequias Junior on 8/8/17.
//  Copyright © 2017 UOL Pagseguro. All rights reserved.
//

#import "SimplePaymentTransactionController.h"
#import "PlugPag.h"
#import "UIUtils.h"

#define MSG_ERROR_DEVICE_NULL   @"Não foi possível efetuar uma conexão com o dispositivo pareado: %i"
#define MSG_ERROR_GENERIC       @"Não foi possível efetuar a operação"

@interface SimplePaymentTransactionController () {
    float amount;
    int cafe;
}

@end

@implementation SimplePaymentTransactionController

#pragma mark - UIViewController Cycle
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    amount = 0;
    cafe = 0;
    [self updateAmount];
    
    [[PlugPag sharedInstance] SetVersionName:@"UnitTest" withVersion:@"V001"];
}

- (void) viewDidAppear:(BOOL)animated {
    
    [super viewDidAppear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - IBAction Buttons
- (IBAction)pagar:(UIButton *)sender {
    
    [sender setEnabled:NO];
    [self updateAmount];
    [UIUtils showProgress];
    
    int ret = [[PlugPag sharedInstance] InitBTConnection];
    
    switch (ret) {
        case RET_OK:
            [self paymentTransaction];
            break;
            
        case ERROR_DEVICE_NULL:
            _txtResult.attributedText = [UIUtils formatTextError:ret withMessage: MSG_ERROR_DEVICE_NULL];
            break;
            
        default:
            _txtResult.attributedText = [UIUtils formatTextError:ret withMessage: MSG_ERROR_GENERIC];
            break;
    }
    
    [UIUtils hideProgress];
    [sender setEnabled:YES];
}

- (IBAction)insertCafe:(UIButton *)sender {
    
    amount = amount + 2.65;
    cafe = cafe + 1;
    [self updateAmount];
}

- (IBAction)deleteCafe:(UIButton *)sender {
    
    if (amount > 0) amount = amount - 2.65;
    if (cafe > 0) cafe = cafe - 1;
    [self updateAmount];
}

#pragma mark - Business Methods
- (void) paymentTransaction {
    
    NSString *value = [NSString stringWithFormat:@"%.02f", amount];
    value = [value stringByReplacingOccurrencesOfString:@"." withString:@""];
    
    int ret = [[PlugPag sharedInstance] SimplePaymentTransaction:CREDIT withInstallmentType:A_VISTA andInstallments:1 andAmount:value andUserReference:@"Pagcafe"];
    
    if (ret == RET_OK) {
        _txtResult.attributedText = [UIUtils formatTextGetLastApprovedTransaction];
    }else{
        _txtResult.attributedText = [UIUtils formatTextError:ret withMessage: MSG_ERROR_GENERIC];
    }
    
    amount = 0;
    cafe = 0;
}

-(void) updateAmount {
    
    if (amount < 1) amount = 2.65;
    if (cafe < 1) cafe = 1;
    
    NSString *msgCafe = cafe < 2 ? @"Café" : @"Cafés";
    
    _txtResult.attributedText = [UIUtils formatCurrency:amount];
    _lblCafe.text = [NSString stringWithFormat:@"%d %@", cafe, msgCafe];
}

@end

File: 1.x/demos/iOS/Example-iOS-Swift/Example-iOS-Swift/SimplePaymentTransactionController.swift
//
//  SimplePaymentTransactionController.swift
//  Example-iOS-Swift
//
//  Created by Hildequias Junior on 8/23/17.
//  Copyright © 2017 UOL Pagseguro. All rights reserved.
//

import UIKit

class SimplePaymentTransactionController: UIViewController {

    @IBOutlet weak var lblCafe: UILabel!
    @IBOutlet weak var txtResult: UITextView!
    @IBOutlet weak var btPayment: UIButton?
    @IBOutlet weak var scInstallmentType: UISegmentedControl!
    
    let MSG_ERROR_DEVICE_NULL   = "Não foi possível efetuar uma conexão com o dispositivo pareado"
    let MSG_ERROR_GENERIC       = "Não foi possível efetuar a operação"
    
    var amount = 0.0
    var cafe = 0
    
    // MARK: - UIViewController Cycle
    override func viewDidLoad() {
        super.viewDidLoad()

        PlugPag.sharedInstance().setVersionName("UnitTest", withVersion: "V001")
        updateAmount()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - IBAction Buttons
    @IBAction func pagar(sender: UIButton) {
        
        sender.isEnabled = false
        updateAmount()
        UIUtils.showProgress()
        
        let ret = PlugPag.sharedInstance().initBTConnection()
        
        switch ret {
        case RET_OK:
            paymentTransaction()
            break
            
        case -2023:
            txtResult?.attributedText = UIUtils.formatTextError(error: Int(ret), msg: "\(MSG_ERROR_DEVICE_NULL)")
            break
            
        default:
            txtResult?.attributedText = UIUtils.formatTextError(error: Int(ret), msg: "\(MSG_ERROR_GENERIC)")
            break
        }
        
        UIUtils.hideProgress()
        sender.isEnabled = true
    }
    
    @IBAction func insertCafe(sender: UIButton) {
    
        amount = amount + 2.65
        cafe = cafe + 1
        updateAmount()
    }
    
    @IBAction func deleteCafe(sender: UIButton) {
    
        if (amount > 0) { amount = amount - 2.65 }
        if (cafe > 0) { cafe = cafe - 1 }
        updateAmount()
    }
    
    // MARK: - Business Methods
    func paymentTransaction() {
        
        var value = String(format: "%.02f", amount)
        value = value.replacingOccurrences(of: ".", with: "")
        
        let paymentMethod = scInstallmentType.selectedSegmentIndex == 0 ? CREDIT : DEBIT
        
        let ret = PlugPag.sharedInstance().simplePaymentTransaction(paymentMethod, withInstallmentType: A_VISTA, andInstallments: 1, andAmount: value, andUserReference: "Pagcafe")
        
        switch ret {
        
            case RET_OK:
                txtResult?.attributedText = UIUtils.formatTextGetLastApprovedTransaction()
                break
        
            case -1019:
                searchResult()
                break
            
            default:
                txtResult?.attributedText = UIUtils.formatTextError(error: Int(ret), msg: "\(MSG_ERROR_GENERIC)")
        }
        
        amount = 0;
        cafe = 0;
    }
    
    func searchResult() {
        
        let ret = PlugPag.sharedInstance().getLastApprovedTransactionStatus()
        
        if ret == RET_OK {
            txtResult?.attributedText = UIUtils.formatTextGetLastApprovedTransaction()
        }else{
            txtResult?.attributedText = UIUtils.formatTextError(error: Int(ret), msg: "\(MSG_ERROR_GENERIC)")
        }
        
    }
    
    func updateAmount() {
        
        if(amount < 1) { amount = 2.65 }
        if(cafe < 1) { cafe = 1 }
        
        let msgCafe = cafe < 2 ? "Café" : "Cafés"
        
        txtResult?.attributedText = UIUtils.formatCurrency(value: Float(amount))
        lblCafe?.text = "\(cafe) \(msgCafe)"
    }
}
